cloud/pkg/cloudhub/servers/httpserver/secretsutil.go

```diff
@@ -27,6 +27,23 @@ const (
 	CloudCoreKeyDataName string = "cloudcorekeydata"
 )
 
+var NamespaceSystem string = constants.KubeEdgeNameSpace
+
+// set namesapce from env CLOUDCORE_POD_NAMESPACE
+func init() {
+	NamespaceSystem = setKubeedgeNamespce(os.Getenv("CLOUDCORE_POD_NAMESPACE"))
+}
+
+func setKubeedgeNamespce(kubeedgeNamespace string) string {
```

Comment:
In which case, `os.Getenv("CLOUDCORE_POD_NAMESPACE")` return empty?

Reply:
If this environment variable is not set, it should should return empty

Reply:
Maybe put it as flag is more helpful, now we do not have configs based on env.

Reply:
> Maybe put it as flag is more helpful, now we do not have configs based on env.

The environment variables CLOUDCORE_POD_NAME and CLOUDCORE_POD_NAMESPACE have been used here to set the ReadinessGate.

>
https://github.com/kubeedge/kubeedge/blob/65b5f8bb40f181cd3cbaaad61c94b34ceb26dac9/cloud/pkg/leaderelection/leaderelection.go#L135
>

Reply:
Ohh, it is to detect running as pod or not, not a config I think. We can discuss it at the following meeting up.